host/src/lib/HelperFunctions.jsx

```diff
@@ -318,23 +319,18 @@ export const checkEqualityWithPrevAnswer = (normValue, prevAnswerValue, prevAnsw
  * @param {IResponse[]} prevAnswer
  * @returns {boolean}
  */
-export const checkEqualityWithOtherAnswers = (rawAnswer, normValue, normType, prevAnswer) => {
-  // loop through each of the normalized answers in each of the previous answers
-  for (let i = 0; i < prevAnswer.normAnswer.length; i++) {
-      if (normType === prevAnswer.normAnswer[i].type && checkEqualityWithPrevAnswer(normValue, prevAnswer.normAnswer[i].value, prevAnswer.normAnswer[i].type)) {
-        return true;
-      }
-      // last ditch check on expressions in case there is a basic match that the type check in the previous conditional misses
-      if ((normType === AnswerType.EXPRESSION || prevAnswer.normAnswer[i].type === AnswerType.EXPRESSION) 
-        && (normValue.toString() === prevAnswer.normAnswer[i].value.toString())) {
-          return true;
-      }
-      // last ditch raw answer checks (5% === 5% enter as an expression, for instance)
-      if (rawAnswer === prevAnswer.value) {
-        return true;
-      }
-  };
-  return false;
+export const checkEqualityWithOtherAnswers = (rawAnswer, normAnswerType, normAnswer,  prevAnswer, correctAnswerRegex) => {
+  // convert to set to optimize the equality check between two arrays
+  // this prevents having to use two nested for/foreach loops etc
+  const prevAnswerSet = new Set(prevAnswer);
+  // expression equality requires a bit more work, as we have to use mathjs to check for symbolic equality
+  if (Number(normAnswerType) === AnswerType.EXPRESSION) {
+    return checkExpressionEquality(normAnswer, prevAnswerSet);
+  }
+  return ( normAnswer.some(item => prevAnswerSet.has(item)) 
+    || (Number(normAnswerType) === AnswerType.STRING && correctAnswerRegex.test(normAnswer))
+    || (rawAnswer === prevAnswer.value) 
+  );
```

Comment:
last two conditions here are final checks if we don't find precise equality using the some/has. This catches substrings within strings as well as a precise match from the rawAnswer